File: scripts/generate_embeddings.py
import os
from openai import OpenAI


api_key = os.getenv("OPENAI_API_KEY")
client = OpenAI(api_key=api_key)


def get_embedding(text, model="text-embedding-3-small"):
    text = text.replace("\n", " ")
    return client.embeddings.create(input=[text], model=model).data[0].embedding


for 

File: scripts/generate_embeddings.ts
import axios from "axios";
import { VectorDB } from 'imvectordb';

const db = new VectorDB();

db.addText("text for semantic search")
// ...add additional text as needed

db.queryText("text to search")


// import { AxiosError } from "axios";


// const generateEmbedding = async (text: string): Promise<number[]> => {
//   try {
//     const response = await axios.post(
//       "https://api.openai.com/v1/embeddings",
//       {
//         input: text,
//         model: "text-embedding-ada-002"
//       },
//       {
//         headers:
//         {
//           "Authorization": `Bearer ${process.env.OPENAI_API_KEY}`,
//           "Content-Type": "application/json"
//         }
//       }
//     );
//   } catch (error) {
//     const axiosError = error as AxiosError;
//     console.error(axiosError);
//   }

//   return response.data.data[0].embedding;
// }